source/qb64pe.bas

```diff
@@ -11767,6 +11767,16 @@ FOR i = 1 TO idn
 
                             unresolved = unresolved + 1
                             sflistn = sflistn + 1
+                            ubound_sf = UBOUND(sfidlist) 'all 3 should have the same limit
+                            IF sflistn > ubound_sf THEN
+                                REDIM _PRESERVE sfidlist(ubound_sf + 1000) AS LONG
+                                REDIM _PRESERVE sfarglist(ubound_sf + 1000) AS INTEGER
+                                REDIM _PRESERVE sfelelist(ubound_sf + 1000) AS INTEGER
+                            END IF
+                            IF ubound_sf > 25000 THEN 'manually set a descriptive error message for the user so they know what's happening.
```

Comment:
Should this rather be `IF sflistn > 25000 THEN` instead of `ubound_sf` ?? Also better move the error checking block before the REDIM block.

As it currently is, it would unnecessarily REDIM the arrays to 26000 elements.

Reply:
Good catch.   And with the misuse of ubound_sf there, it'd never have a chance to STOP erroring out.   sflistn resets itself and rebuilds itself with each new pass of the IDE (so if you insert a SUB between subs, it'll take the old id numbers from them and try and keep sub numbering in order), so when someone removes a SUB, that value *should* drop by one, in total.   The idea here is for the IDE to just toss an error message, without crashing, and which the user could then consolidate subs/functions (or remove unneeded ones), to get back below that 25,000 limit and stop the error from ever occurring.

Honestly though, I can't hardly imagine that anyone would *ever* have 25,000 subs and functions in a program.  From the day QB64 was first conceived, to now, Grymm has been the only person to ever report the glitch with a cap of just 1,000.   I truly think if that cap is ever reached, it's going to be because of some endless loop on our part in the IDE, and not something the user can ever fix to begin with.  :)